### DISCUSSION TITLE
Check git config for unsupported options before running

### DISCUSSION THREAD

**Discussion by mattrq** [arxanas/git-branchless#532] [Ideas]
Git branchless does not support all the options that git supports e.g.(`core.splitIndex` and `extensions.worktreeConfig`). However, these can cause git branchless to fail during running (but not always at the start).

For known options incompatible with git branchless, I suggest a check and a clear warning (or prompt to fix) when any branchless command runs (assuming that they are quick to do). 

Many of these are optimisations for mono-repos that branchless help with in a different way.

I also think this will help the adoption of git branchless as it will reduce the barrier to entry/perceived flakiness.

**Comment by arxanas** [comment]
Hi @mattrq, thanks for the feedback! I opened https://github.com/arxanas/git-branchless/pull/543 to warn about `extensions.worktreeConfig`. I tried a few operations with `core.splitIndex` set, but they didn't fail. Can you give an example of some situations where it causes an issue?
